changeset:   623785:a6b80ee8c35f
tag:         tip
parent:      623782:b1ed2fa50612
user:        stransky <stransky@redhat.com>
date:        Wed Jun 15 14:54:19 2022 +0200
files:       dom/media/platforms/ffmpeg/FFmpegVideoDecoder.cpp
description:
Bug 1758948 [FFmpeg] Use AVFrame::pts instead of AVFrame::pkt_pts on ffmpeg 4.x r?alwu

AVFrame::pkt_pts has been deprecated and gives us wrong values for AV1 VA-API. Let's use AVFrame::pts instead on ffmpeg 4.x
as well as we use on ffmpeg 5.0 where AVFrame::pkt_pts is removed.

Differential Revision: https://phabricator.services.mozilla.com/D149386


diff --git a/dom/media/platforms/ffmpeg/FFmpegVideoDecoder.cpp b/dom/media/platforms/ffmpeg/FFmpegVideoDecoder.cpp
--- a/dom/media/platforms/ffmpeg/FFmpegVideoDecoder.cpp
+++ b/dom/media/platforms/ffmpeg/FFmpegVideoDecoder.cpp
@@ -774,7 +774,7 @@ void FFmpegVideoDecoder<LIBAV_VER>::Init
 #endif
 
 static int64_t GetFramePts(AVFrame* aFrame) {
-#if LIBAVCODEC_VERSION_MAJOR > 58
+#if LIBAVCODEC_VERSION_MAJOR > 57
   return aFrame->pts;
 #else
   return aFrame->pkt_pts;

